aptos-move/move-examples/token_objects/ambassador/client/ambassador.ts

```diff
@@ -0,0 +1,171 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { AptosAccount, AptosClient, HexString } from "aptos";
+
+const NODE_URL = process.env.APTOS_NODE_URL || "https://fullnode.testnet.aptoslabs.com";
+const FAUCET_URL = process.env.APTOS_FAUCET_URL || "https://faucet.testnet.aptoslabs.com";
+
+function fetchGraphQL(
+    operationsDoc: string,
+    operationName: string,
+    variables: Record<string, any>
+) {
+    return fetch('https://indexer-testnet.staging.gcp.aptosdev.com/v1/graphql', {
+        method: 'POST',
+        body: JSON.stringify({
+            query: operationsDoc,
+            variables,
+            operationName,
+        }),
+    }).then(result => result.json());
+}
+
+async function getTokenAddr(ownerAddr: HexString, tokenName: string): Promise<HexString> {
+    const operation = `
+  query TokenOwnership {
+    current_token_ownerships_v2(
+      where: {owner_address: {_eq: "${ownerAddr}"}, amount: {_gt: 0}}
+      order_by: {last_transaction_version: desc}
+    ) {
+      amount
+      current_token_data {
+        token_uri
+        token_data_id
+        token_properties
+        token_name
+        current_collection {
+          collection_name
+        }
+      }
+      token_standard
+    }
+  }
+`;
+    const data = await fetchGraphQL(operation, "TokenOwnership", {});
+    const tokenOwnership = data.data;
+
+    // TODO: `tokenOwnership` can be obtained by the following code when a new SDK version is ready.
+    // const tokenOwnership = await provider.getTokenOwnershipV2(ownerAddr);
+
+    for (const ownership of tokenOwnership.current_token_ownerships_v2) {
+        if (ownership.current_token_data.token_name === tokenName) {
```

Comment:
Can we allow filtering by token name in the query? cc @0xmaayan @bowenyang007

Reply:
SDK can provide something like `getToken(owner, tokenName)` ?  @bowenyang007 any concerns about performance?